tests/bsim/bluetooth/host/adv/chain/src/main.c

```diff
@@ -94,25 +98,29 @@ static void scan_recv(const struct bt_le_scan_recv_info *info,
 
 	data_len = buf->len;
 	if (data_len != DATA_LEN) {
+		printk("Unexpected length\n");
```

Comment:
Shouldn't this fail? Please also print the `data_len` here for debugging purposes

Reply:
Didn't cause fail in the original test. Text updated to better reflect this, and print more debugging info